x/staking/types/validator.go

```diff
@@ -247,7 +256,11 @@ func (v Validator) ABCIValidatorUpdate() abci.ValidatorUpdate {
 // ABCIValidatorUpdateZero returns an abci.ValidatorUpdate from a staking validator type
 // with zero power used for validator updates.
 func (v Validator) ABCIValidatorUpdateZero() abci.ValidatorUpdate {
-	pk, err := encoding.PubKeyToProto(v.GetConsPubKey())
+	consPk, err := v.GetConsPubKey()
+	if err != nil {
+		panic(err)
+	}
+	pk, err := encoding.PubKeyToProto(consPk)
```

Comment:
We should be careful here. After this PR, `v.GetConsPubKey()` is always our Pubkey, and `encoding.PubKeyToProto` expect a TM Pubkey. We need to call `IntoTmPubKey` before.

In general, what I think is the cleanest is:
- in our codebase, manipulate our own PubKey
- when we need to interface with TM (e.g. here with the ABCI), we convert using `IntoTmPubKey`
- all this should be enforced statically at compile time with #7419

Reply:
`Validator.GetConsPubKey` already converts to `tmcrypto.ed25519`.  I don't think we need to change anything here. Please confirm.

Reply:
Ah you're right, then we don't need to change anything here.

We could maybe rename it to `.GetTmConsPubKey()`, what do you think? I initially thought it was a getter for the `.ConsesusPubKey` field.

Reply:
The method returns tm key only if the `IntoTmPubKey` is implemented for that key.

Reply:
All our pubkeys that can be converted to a TM pubkey implement `IntoTmPubKey`. So
- if `pk` is our pubkey, convert to TM and return
- if `pk` is already a TM pubkey, return it.

In every case, the return value is a TM pubkey.

Note: I still don't know how the case "If `pk` is already a TM pubkey" can even happen, hence the comment below. But I think I saw it once or twice in tests.

Reply:
If we want to assure that `tmcrypto` PubKey is returned then we need to do it in `GetConsPubKey` and return an error if for some reason it's not convertible to TM PubKey. 

I will update the method and make that assertion.

Reply:
I renamed to `TmConsPubKey`